primitive_function.go

```diff
@@ -77,29 +77,99 @@ func (self *PrimitiveFunction) parseNumArgs(argCount string) {
 		}
 
 		var intTerm int
```

Comment:
We should definitely have comments in this function as to why those `is*` functions exist and leave links to the Go issues.

Reply:
I added a comment explaining the reason for these new functions, with links to both related issues.